main/lsp/requests/completion.cc

```diff
@@ -838,8 +871,28 @@ vector<unique_ptr<CompletionItem>> CompletionTask::getCompletionItems(LSPTypeche
     // ----- locals -----
 
     vector<core::NameRef> similarLocals;
+    bool localsAreFields = false;
     if (params.enclosingMethod.exists()) {
-        auto locals = localNamesForMethod(typechecker, params.enclosingMethod, params.queryLoc);
+        vector<core::NameRef> locals;
+
+        // Slyly reuse `UnresolvedIdent::Kind` to both determine what kind of
+        // thing we're going to find and to determine the search space for
+        // `fieldsForClass`.
+        auto kind = ast::UnresolvedIdent::Kind::Local;
+        if (params.prefix.size() >= 2 && absl::StartsWith(params.prefix, "@@")) {
+            kind = ast::UnresolvedIdent::Kind::Class;
+        } else if (params.prefix.size() >= 1 && absl::StartsWith(params.prefix, "@")) {
+            kind = ast::UnresolvedIdent::Kind::Instance;
+        }
```

Comment:
If we always have to compute this anyways, let's just add this field to the `IdentResponse`, because it should always be known at the point where the `IdentResponse` is pushed.

Reply:
I think overloading `UnresolvedIdent::Kind` here is appropriate, if a bit hacky; I don't think putting `UnresolvedIdent::Kind` in `IdentResponse` is a good thing to do.  Unless you mean essentially duplicating `UnresolvedIdent::Kind` in `IdentResponse`, which might not be a bad idea.  (Though if we did some kind of symbol table search, then we wouldn't need to compute this bit.)